### ISSUE TITLE
[BUG] note blocks and RestrictedCreative 

### ISSUE THREAD

**Issue by 100petr** [PluginBugs/Issues-ItemsAdder#866]
**Describe the bug**
I have the RestrictedCreative plugin (https://www.spigotmc.org/resources/restrictedcreative.42790/) on the server, but if I add ItemsAdder, the players place blocks of note block and then destroy these blocks and the block drop.

**To Reproduce**
1. Add ItemsAdder and RestrictedCreative on the server
2. Place note blocks
3. Put blocks next to them
4. Destroy these blocks

**Expected behavior**
Blocks do not drop when placed next to a note block

**Screenshots**
Video: https://youtu.be/nmkrk9ZuVLM

**Info (please complete the following information):**
 - Server version: Purpur-1065 (MC: 1.16.5)
 - Client version: 1.16.5
 - Plugin version: 2.3.5

**Additional context**
I think it will have something to do with the REAL_NOTE blocks feature. If I remove ItemsAdder, the blocks will not drop.

**Comment by LoneDev6** [comment]
So basically you set RestrictedCreative to disable drops in the world but if you break a normal block which is near a custom block you get the drop?

**Comment by 100petr** [comment]
The plugin prevents dropping items built in creative. If the block is placed next to the note block and the ItemsAdder is installed, the block will drop.

**Comment by LoneDev6** [comment]
But it won't drop if you place the item not near the noteblock, right?

**Comment by 100petr** [comment]
Yes. One block was skipped between the noteblock and the player gained nothing. 
![image](https://user-images.githubusercontent.com/29830635/116582212-6b079b00-a915-11eb-88ab-8b3d6cdc9810.png)

**Comment by LoneDev6** [comment]
mhhh this is very strange btw, I don't think I have a way to fix this but I will keep the issue open

**Comment by LoneDev6** [comment]
If you break in creative will it drop the item or not?
Does this happen only in survival mode? as you shown in the video

**Comment by 100petr** [comment]
Nothing drops if destruction is in creative, only in survival.

čt 29. 4. 2021 v 19:57 odesílatel LoneDev ***@***.***> napsal:

**Comment by LoneDev6** [comment]
Why your players have access to `/gamemode survival` command?
Can't you allow your users to stay only in creative in this world?
Also you could disable drops completely in this world.

I don't think I can easily fix this issue sincerely simply because I can't identify the cause of this problem, very strange

**Comment by 100petr** [comment]
I tried and found this:
2.2.1 is not this problem
2.2.2 is this problem

**Comment by LoneDev6** [comment]
So 2.2.2 introduced the issue?

**Comment by 100petr** [comment]
I apologize for the incorrect information, we try it all the time (except for short breaks) and in the end I came up with this:


https://user-images.githubusercontent.com/29830635/117183878-5ecd8300-add8-11eb-85f4-b3d9627eb357.mp4


https://user-images.githubusercontent.com/29830635/117185009-aa346100-add9-11eb-8e82-02b08bbdb670.mp4


The recording gets stuck, sometimes I have to change the recording quality
So my definitive finding: IA2.1.42 - no problem ; IA2.2.1 problem

**Comment by 100petr** [comment]
If I switched disable-REAL_NOTE-behavior__DISABLE_AT_YOUR_OWN_RISK to true for the test, then there is no problem. They just look noteblock like xray :D 
![image](https://user-images.githubusercontent.com/29830635/117495434-e8668780-af75-11eb-8f3c-b008d045fcfa.png)

**Comment by LoneDev6** [comment]
I sincerely don't think I can fix this, anyway I don't own a copy of RestrictedCreative so I can't test this.
If you want you can send me a copy of the plugin on Discord

**Comment by LoneDev6** [comment]
Found the cause.
It was because RestrictedCreative checks type and if it's still AIR it will be ignored.
Sometimes it's AIR because the server didn't manage to place the block in time, some parts of ItemsAdder are async so this is the cause.
I had to introduce a workaround and I hope it won't cause performance drops...

This is the RestrictedCreative code line that caused the problem:
![immagine](https://user-images.githubusercontent.com/27242001/117678769-fbfd3280-b1af-11eb-8e67-352916f9f85d.png)
